### PR TITLE
Drop `Kokkos::Array` specialization of `Impl::ViewMapping`

### PR DESCRIPTION
**Pull request by dalg24**
This is a **breaking change** so it needs to be considered carefully.

View currently behave differently when the value type is `Kokkos::Array`.  Essentially it is treated as an extra compile-time dimension in the view.  It came up recently in #6883 where one reported that constructing an unmanaged view of arrays would not compile.  Here I suggest that we get rid of that special treatment because it is source of extra complexity and it makes the upcoming `mdpsan` view refactor more complex.  I could not find any use of view of arrays in Trilinos (admittedly just grepped quickly through the code base, not a throrough search).

Feel free to review or comment to express your concerns.  My intent is to discuss this PR at the next developer meeting and see if we can reach consensus to remove.

### CODE REVIEW

**Comment by masterleinad** [comment]
I like removing these specializations but didn't want to break backward compatibility. I'm curious to here about the rationale for the current design.

**Comment by crtrott** [comment]
I am ok with pulling this out.

**Comment by tpadioleau** [comment]
Out of curiosity, is there still a necessity for the alias `specialize` ? Was `Kokkos::Array<>` the only reason for it ?

**Comment by masterleinad** [comment]
> Out of curiosity, is there still a necessity for the alias `specialize` ? Was `Kokkos::Array<>` the only reason for it ?

It's also `Stokhos`/`Sacado`.

**Comment by ldh4** [comment]
I'm also for dropping this specialization. It's a very subtle and specific detail in behavior that doesn't seem to be documented either.

> I'm curious to here about the rationale for the current design.

I'm also curious about this. Presumably, there was a need for this specialization at some point?

**Comment by dalg24** [comment]
(rebased and fix conflicts following the merge of #6905 no actual changes)

File: core/unit_test/TestOther.hpp [hunk 1]
```diff
@@ -16,7 +16,6 @@
 
 #ifndef KOKKOS_TEST_OTHER_HPP
 #define KOKKOS_TEST_OTHER_HPP
-#include <TestAggregate.hpp>
```
Comment (aprokop) [inline core/unit_test/TestOther.hpp:19]: Should `TestAggregate.hpp` file be also removed?

Comment (dalg24) [reply core/unit_test/TestOther.hpp:19]: Sloppy rebase.  Fixed.  Thanks.

**Comment by crtrott** [comment]
Note with the move to mdspan there is a different way to achieve the desired behavior in Sacado/Stokhos, which I have previously prototyped using accessors. We do not need this awkward extra hidden dimension. So with the move to mdspan the specialize thing goes away.

**Review by crtrott — APPROVED** [review]

**Comment by dalg24** [comment]
Ignoring unrelated runtime failures in one of the HIP builds

**Pull request merged**